packages/contracts-bedrock/src/L1/OPPrestateUpdater.sol

```diff
@@ -0,0 +1,177 @@
+        for (uint256 i = 0; i < _prestateUpdateInputs.length; i++) {
+            IFaultDisputeGame permissionlessDisputeGame = IFaultDisputeGame(
+                address(
+                    getGameImplementation(
+                        IDisputeGameFactory(_prestateUpdateInputs[i].systemConfigProxy.disputeGameFactory()),
+                        GameTypes.CANNON
+                    )
+                )
+            );
+
+            bool hasFDG = !(address(permissionlessDisputeGame) == address(0));
+
+            AddGameInput[] memory inputs = new AddGameInput[](hasFDG ? 2 : 1);
+            AddGameInput memory pdgInput;
+            AddGameInput memory fdgInput;
+
+            if (Claim.unwrap(_prestateUpdateInputs[i].absolutePrestate) == bytes32(0)) {
+                revert PDGPrestateRequired();
+            }
+            // Get the current game implementation to copy parameters from
+            IDisputeGameFactory dgf =
+                IDisputeGameFactory(_prestateUpdateInputs[i].systemConfigProxy.disputeGameFactory());
+            IPermissionedDisputeGame pdg =
+                IPermissionedDisputeGame(address(getGameImplementation(dgf, GameTypes.PERMISSIONED_CANNON)));
+            uint256 initBond = dgf.initBonds(GameTypes.PERMISSIONED_CANNON);
+
+            // Get the existing game parameters
+            IFaultDisputeGame.GameConstructorParams memory pdgParams =
+                getGameConstructorParams(IFaultDisputeGame(address(pdg)));
+
+            string memory saltMixer =
+                string.concat("prestate_update", string(abi.encode(_prestateUpdateInputs[i].systemConfigProxy)));
+            // Create game input with updated prestate but same other params
+            pdgInput = AddGameInput({
+                disputeAbsolutePrestate: _prestateUpdateInputs[i].absolutePrestate,
+                saltMixer: saltMixer,
+                systemConfig: _prestateUpdateInputs[i].systemConfigProxy,
+                proxyAdmin: _prestateUpdateInputs[i].proxyAdmin,
+                delayedWETH: IDelayedWETH(payable(address(pdgParams.weth))),
+                disputeGameType: pdgParams.gameType,
+                disputeMaxGameDepth: pdgParams.maxGameDepth,
+                disputeSplitDepth: pdgParams.splitDepth,
+                disputeClockExtension: pdgParams.clockExtension,
+                disputeMaxClockDuration: pdgParams.maxClockDuration,
+                initialBond: initBond,
+                vm: pdgParams.vm,
+                permissioned: true
+            });
+
+            // If fault dispute prestate is provided, create a new game with the same parameters but updated prestate
+            if (hasFDG) {
+                // Get the current game implementation to copy parameters from
+                IFaultDisputeGame fdg = IFaultDisputeGame(address(getGameImplementation(dgf, GameTypes.CANNON)));
+                if (address(fdg) == address(0)) revert FDGNotFound();
```

Comment:
This should no longer be reachable, since we updated the definition of `hasFDG` to be the opposite of this condition.

Reply:
Agreed, and the line above it (141) is a duplicate of line 90. We should delete both these lines, and rename the variable on line 90 from `permissionlessDisputeGame` to `fdg` for consistency (I prefer using names of `fdg` and `pdg` because that feels more readable than `permissionedDisputeGame` and `permissionlessDisputeGame` which have too many identical characters lol)